src/RTC_PCF8523.cpp

```diff
@@ -1,292 +1,332 @@
+}
+
+/**************************************************************************/
+/*!
+    @brief  Stop all timers, clear their flags and settings on the PCF8523.
+    @details This includes the Countdown Timer, Second Timer, and any CLKOUT
+   square wave configured with writeSqwPinMode().
+*/
+/**************************************************************************/
+void RTC_PCF8523::deconfigureAllTimers() {
+  disableSecondTimer(); // Surgically clears CONTROL_1
+  write_register(PCF8523_CONTROL_2, 0);
+  write_register(PCF8523_CLKOUTCONTROL, 0);
+  write_register(PCF8523_TIMER_B_FRCTL, 0);
+  write_register(PCF8523_TIMER_B_VALUE, 0);
+}
+
+/**************************************************************************/
+/*!
+    @brief Compensate the drift of the RTC.
+    @details This method sets the "offset" register of the PCF8523,
+      which can be used to correct a previously measured drift rate.
+      Two correction modes are available:
+
+      - **PCF8523\_TwoHours**: Clock adjustments are performed on
+        `offset` consecutive minutes every two hours. This is the most
+        energy-efficient mode.
+
+      - **PCF8523\_OneMinute**: Clock adjustments are performed on
+        `offset` consecutive seconds every minute. Extra adjustments are
+        performed on the last second of the minute is `abs(offset)>60`.
+
+      The `offset` parameter sets the correction amount in units of
+      roughly 4&nbsp;ppm. The exact unit depends on the selected mode:
+
+      |  mode               | offset unit                            |
+      |---------------------|----------------------------------------|
+      | `PCF8523_TwoHours`  | 4.340 ppm = 0.375 s/day = 2.625 s/week |
+      | `PCF8523_OneMinute` | 4.069 ppm = 0.352 s/day = 2.461 s/week |
+
+      See the accompanying sketch pcf8523.ino for an example on how to
+      use this method.
+
+    @param mode Correction mode, either `PCF8523_TwoHours` or
+      `PCF8523_OneMinute`.
+    @param offset Correction amount, from -64 to +63. A positive offset
+      makes the clock slower.
+*/
+/**************************************************************************/
+void RTC_PCF8523::calibrate(Pcf8523OffsetMode mode, int8_t offset) {
+  write_register(PCF8523_OFFSET, ((uint8_t)offset & 0x7F) | mode);
+}
+/**************************************************************************/
+/*!
+    @brief read the offset register
+*/
+/**************************************************************************/
+
+uint8_t RTC_PCF8523::readOffsetReg() {
+  uint8_t OffsetReg = read_register(PCF8523_OFFSET);
+  return OffsetReg;
+}
+
+/**************************************************************************/
+/*!
+    @brief read the offset register and return OffsetMode
+*/
+/**************************************************************************/
+
+String RTC_PCF8523::getOffsetMode() {
+  String OffsetMode;
+  if bitRead (readOffsetReg(), 7) {

```

Comment:
Just noticed this: the condition of the `if` should be in parentheses:
```suggestion
  if (bitRead(readOffsetReg(), 7)) {
```